src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs

```diff
@@ -89,6 +89,39 @@ public async Task<Response<ConfigurationSetting>> AddAsync(ConfigurationSetting
             }
         }
 
+        public async Task<Response<ConfigurationSetting>> AddAsync(string key, string value, CancellationToken cancellation = default)
+        {
+            if (string.IsNullOrEmpty(key)) throw new ArgumentNullException($"{nameof(key)}");
+            if (string.IsNullOrEmpty(value)) throw new ArgumentNullException($"{nameof(value)}");
```

Comment:
Is an empty value acceptable?

Reply:
hmm good question. I don't see why not. At the same time why would someone want to only create only a key? with nothing else?

Reply:
As long as the service allows it I would think we would also want to allow it. @KrzysztofCwalina what do you think?

Reply:
I think we should allow empty strings, if the service allows them.

Reply:
Ok. Will change